ui/lib/core/addon/components/search-select.js

```diff
@@ -103,77 +114,103 @@ export default Component.extend({
         ...(this.idKey !== 'id' && this.customizeObject(matchingOption)),
       };
     });
-    this.set('selectedOptions', formattedOptions);
-    if (this.options) {
-      options = this.options.concat(options).uniq();
+  }
+
+  @task
+  *fetchOptions() {
+    if (this.args.parentManageSelected) {
+      this.selectedOptions = this.args.parentManageSelected;
     }
-    this.set('options', options);
-  },
-  fetchOptions: task(function* () {
-    if (!this.models) {
-      if (this.options) {
-        this.formatOptions(this.options);
+    this.dropdownOptions = [];
+    if (!this.args.models) {
+      if (this.args.options) {
+        const { options } = this.args;
+        if (options.some((e) => Object.keys(e).includes('groupName'))) {
+          this.dropdownOptions = options;
+        } else {
+          this.dropdownOptions = [...this.addSearchText(options)];
+        }
       }
       return;
     }
-    if (this.idKey !== 'id') {
-      // if passing a dynamic idKey, then display it in the dropdown beside the name
-      this.set('shouldRenderName', true);
-    }
-    for (let modelType of this.models) {
-      if (modelType.includes('identity')) {
-        this.set('shouldRenderName', true);
-      }
+
+    for (let modelType of this.args.models) {
       try {
         let queryOptions = {};
-        if (this.backend) {
-          queryOptions = { backend: this.backend };
+        if (this.args.backend) {
+          queryOptions = { backend: this.args.backend };
         }
-        if (this.queryObject) {
-          queryOptions = this.queryObject;
+        if (this.args.queryObject) {
+          queryOptions = this.args.queryObject;
         }
-        let options = yield this.store.query(modelType, queryOptions);
-        this.formatOptions(options);
+        this.allOptions = yield this.store.query(modelType, queryOptions);
+        this.dropdownOptions = [...this.dropdownOptions, ...this.addSearchText(this.allOptions)];
+        this.selectedOptions = this.formatSelectedAndUpdateDropdown(this.selectedOptions);
       } catch (err) {
         if (err.httpStatus === 404) {
-          if (!this.options) {
```

Comment:
I confess I kind of lost track of `this.options` at this point and am not confident I understand what this `if` block is checking for...that nothing has been set at all? 

The new value of `this.dropdownOptions` will always be truthy, since I 'reset' it to an empty array at the beginning of this method.